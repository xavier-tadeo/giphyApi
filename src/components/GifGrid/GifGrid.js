import React from "react";

export const GifGrid = ({ categorie }) => {
  const getGif = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=dragon+ball&limit=10&api_key=z3bZoUQrp7F5qjY3Xr5MUVKBEJZsikUw";
    const response = await fetch(url);
    const { data } = await response.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        img: img.images.downsized_medium.url,
        title: img.title,
      };
    });
    console.log(gifs);
  };

  getGif();
  return <li>{categorie}</li>;
};
