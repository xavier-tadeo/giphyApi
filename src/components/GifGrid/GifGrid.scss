.container__card {
  display: flex;
  flex-flow: wrap;
  border-radius: 40px;
  margin: 10px;
}
