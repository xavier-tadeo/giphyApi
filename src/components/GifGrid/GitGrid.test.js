import { GifGrid } from "./GifGrid";
import { shallow } from "enzyme";

describe("Given a GitGrid component", () => {
  describe("When it render", () => {
    test("It should rendered a liTag with name 'categorie' and one divTag", () => {
      const categorie = "Ball dragon";

      const wrapper = shallow(<GifGrid categorie={categorie} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
