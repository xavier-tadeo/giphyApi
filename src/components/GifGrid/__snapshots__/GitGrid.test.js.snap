// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a GitGrid component When it call the function 'useFetchGif' It should show images 1`] = `
<Fragment>
  <li
    className="animate__animated animate__bounce"
  >
    Ball dragon
  </li>
  <div
    className="container__card"
  >
    <GitGridItem
      id="ABC"
      key="ABC"
      title="Algo"
      url="url"
    />
  </div>
</Fragment>
`;

exports[`Given a GitGrid component When it render It should rendered a liTag with name 'categorie' and one divTag 1`] = `
<Fragment>
  <li
    className="animate__animated animate__bounce"
  >
    Ball dragon
  </li>
  <div
    className="container__card"
  >
    <p
      className="animate__animated animate__flash"
    >
      "Loading..."
    </p>
  </div>
</Fragment>
`;
